src/Http/GraphClientFactory.php

```diff
@@ -133,41 +158,42 @@ public function sendAsyncRequest(RequestInterface $request): Promise {
         };
     }
 
+    /**
+     * Return default handler stack for Graph
+     *
+     * @param null $handler final handler
+     * @return HandlerStack
+     */
+    public static function getDefaultHandlerStack($handler = null): HandlerStack
+    {
+        $handler = ($handler) ?: Utils::chooseHandler();
+        $handlerStack = new HandlerStack($handler);
+        $handlerStack->push(GraphMiddleware::retry());
+        $handlerStack->push(GuzzleMiddleware::redirect());
+        $handlerStack->push(GraphMiddleware::graphTelemetry(self::$graphTelemetryOption));
```

Comment:
We do need to make sure that we don't apply telemetry if the request is a retry.

Reply:
don't set SdkVersion & client-request-id at all OR ["don't append new values"](https://github.com/microsoftgraph/msgraph-sdk-design/pull/75/files#diff-6fd6f9a6a2dde99527045dd50b792a442507017c6e4fb257a2c3009750fdd781R26) if the initial request already contained the telemetry headers?

Reply:
Don't append new values to SdkVersion header if the request is a retry. All the existing values should still be valid, and the retry handler is managing retry-attempt.

My thinking is that we don't reset client-request-id as far as the client is concerned; it is the same request when it's a retry.

Reply:
Got it.

Issue is that for PHP, we can't retrieve the request from a response object. So the Retry Handler holds onto the request status as it was before any other middleware down the chain [See here](https://github.com/microsoft/kiota/blob/main/http/php/guzzle/src/Middleware/RetryHandler.php#L64).

Therefore a retry request will NOT contain any Telemetry headers and thus will need to be appended.
Good thing is that we're guaranteed that the Telemetry headers will be same as the previous request since there's no room to update the request options at the retry handler level